### DISCUSSION TITLE
Index out of bounds ERROR

### DISCUSSION THREAD

**Discussion by imariaguiar** [Parcels-code/Parcels#1514] [Q&A] [answered]
Hello there, I was running this 3D simulation for the years 2008 to 2021, everything ran fine until the year of 2018, but starting from the year 2019, I received an error. I've already checked my velocity field data and everything seems okay. Does anyone have any ideas on what I might be doing wrong? 

This is my code for the pset and execution, below is the error I get when running from 2019 onwards.

npart_lat = 25
npart_depth = 30

lons = np.linspace(-23, -23, npart_lat)
lats = np.linspace(1, -1, npart_lat)
depths = np.linspace(0, 290, npart_depth)

periodos = ['DJF','MAM','JJA','SON']

for ano in range(2019,2022):
    for periodo in periodos:
        if  periodo == 'DJF':
            start_date = datetime.date(ano - 1, 12, 1)
        elif periodo == 'MAM':
            start_date = datetime.date(ano, 3, 1)
        elif periodo == 'JJA':
            start_date = datetime.date(ano, 6, 1)
        elif periodo == 'SON': 
            start_date = datetime.date(ano, 9, 1)
            
        output_file_name = f'euc_sec_{periodo}_{ano}.zarr'
        
        pset = ParticleSet(fieldset=fieldset,
                           pclass=ScipyParticle,
                           lon=np.repeat(lons, npart_depth),
                           lat=np.repeat(lats, npart_depth),
                           depth=np.tile(depths, npart_lat),
                           time=start_date)
        
        output_file = pset.ParticleFile(name=output_file_name, outputdt=delta(hours=24))

        pset.execute(AdvectionRK4_3D,
                     runtime=delta(days=90),
                     dt=delta(hours=6),
                     moviedt=delta(hours=24),
                     output_file=output_file,
                     recovery={ErrorCode.ErrorOutOfBounds: DeleteParticle})  

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

IndexError                                Traceback (most recent call last)
Cell In[70], line 37
     34 output_file = pset.ParticleFile(name=output_file_name, outputdt=delta(hours=24))
     36 # executando a simulação
---> 37 pset.execute(AdvectionRK4_3D,
     38              runtime=delta(days=90),
     39              dt=delta(hours=6),
     40              moviedt=delta(hours=24),
     41              output_file=output_file,
     42              recovery={ErrorCode.ErrorOutOfBounds: DeleteParticle})       

File ~\anaconda3\envs\py3_parcels\lib\site-packages\parcels\particleset\baseparticleset.py:427, in BaseParticleSet.execute(self, pyfunc, pyfunc_inter, endtime, runtime, dt, moviedt, recovery, output_file, movie_background_field, verbose_progress, postIterationCallbacks, callbackdt)
    425 next_movie = time + moviedt if dt > 0 else time - moviedt
    426 next_callback = time + callbackdt if dt > 0 else time - callbackdt
--> 427 next_input = self.fieldset.computeTimeChunk(time, np.sign(dt)) if self.fieldset is not None else np.inf
    429 tol = 1e-12
    430 if verbose_progress is None:

File ~\anaconda3\envs\py3_parcels\lib\site-packages\parcels\fieldset.py:1079, in FieldSet.computeTimeChunk(self, time, dt)
   1077             fb.close()
   1078         fb = None
-> 1079     data = f.computeTimeChunk(data, tind)
   1080 data = f.rescale_and_set_minmax(data)
   1082 if (isinstance(f.data, DeferredArray)):

File ~\anaconda3\envs\py3_parcels\lib\site-packages\parcels\field.py:1394, in Field.computeTimeChunk(self, data, tindex)
   1391     timestamp = self.timestamps[np.where(ti < summedlen)[0][0]]
   1393 rechunk_callback_fields = self.chunk_setup if isinstance(tindex, list) else None
-> 1394 filebuffer = self._field_fb_class(self.dataFiles[g.ti + tindex], self.dimensions, self.indices,
   1395                                   netcdf_engine=self.netcdf_engine, timestamp=timestamp,
   1396                                   interp_method=self.interp_method,
   1397                                   data_full_zdim=self.data_full_zdim,
   1398                                   chunksize=self.chunksize,
   1399                                   rechunk_callback_fields=rechunk_callback_fields,
   1400                                   chunkdims_name_map=self.netcdf_chunkdims_name_map,
   1401                                   netcdf_decodewarning=self.netcdf_decodewarning)
   1402 filebuffer.__enter__()
   1403 time_data = filebuffer.time

IndexError: index 4352 is out of bounds for axis 0 with size 4352

**Comment by erikvansebille** [comment, marked as answer]
Dear @imariaguiar, thanks for using Parcels and for reporting this problem. From the error message, it seems that something went wrong with the time dimension when creating your `fieldset`. Did you use `timestamps`, by any chance? Could it be that one variables has fewer time-slices than another?

By the way, it seems that you use a very old version of Parcels; is there a reason for that? We are now at v3.0.2; I recommend running with the latest version as much as possible. Note that e.g. `recovery` has changed after v3, see https://docs.oceanparcels.org/en/latest/examples/tutorial_kernelloop.html#Working-with-Status-Codes
